prometheus-connector/src/main/java/org/apache/flink/connector/prometheus/sink/PrometheusTimeSeries.java

```diff
@@ -0,0 +1,186 @@
+ * labels. The other labels, as key/value, are appended after the '__name__' label.
+ */
+@Public
+public class PrometheusTimeSeries implements Serializable {
+    /** A single Label. */
+    public static class Label implements Serializable {
+        private final String name;
+        private final String value;
+
+        public Label(String name, String value) {
+            this.name = name;
+            this.value = value;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public String getValue() {
+            return value;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            Label label = (Label) o;
+            return new EqualsBuilder()
+                    .append(name, label.name)
+                    .append(value, label.value)
+                    .isEquals();
+        }
+
+        @Override
+        public int hashCode() {
+            return new HashCodeBuilder(17, 37).append(name).append(value).toHashCode();
+        }
+    }
+
+    /** A single Sample. */
+    public static class Sample implements Serializable {
+        private final double value;
+        private final long timestamp;
+
+        public Sample(double value, long timestamp) {
+            this.value = value;
+            this.timestamp = timestamp;
+        }
+
+        public double getValue() {
+            return value;
+        }
+
+        public long getTimestamp() {
+            return timestamp;
+        }
+    }
+
+    private final Label[] labels;
+    private final Sample[] samples;
+    private final String metricName;
+
+    public PrometheusTimeSeries(String metricName, Label[] labels, Sample[] samples) {
+        this.metricName = metricName;
+        this.labels = labels;
+        this.samples = samples;
+    }
+
+    public Label[] getLabels() {
+        return labels;
+    }
+
+    public Sample[] getSamples() {
+        return samples;
+    }
+
+    public String getMetricName() {
+        return metricName;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static Builder builderFrom(PrometheusTimeSeries other) {
+        return new Builder(
+                Arrays.asList(other.labels), Arrays.asList(other.samples), other.metricName);
+    }
+
+    /** Builder for sink input pojo instance. */
+    public static final class Builder {
+        private List<Label> labels = new ArrayList<>();
+        private List<Sample> samples = new ArrayList<>();
+        private String metricName;
+
+        private Builder(List<Label> labels, List<Sample> samples, String metricName) {
+            this.labels = labels;
+            this.samples = samples;
+            this.metricName = metricName;
+        }
```

Comment:
Can we add unit test coverage for this? 
<img width="717" alt="image" src="https://github.com/apache/flink-connector-prometheus/assets/35062175/e03c8c41-bf9b-4339-ab0f-34ac4b62b241">

Reply:
I assume you mean testing the builder, not its constructor.
Added a test for PrometheusTimeSeries.Builder